pinot-spi/src/main/java/org/apache/pinot/spi/data/readers/RecordExtractorConfig.java

```diff
@@ -18,8 +18,13 @@
  */
 package org.apache.pinot.spi.data.readers;
 
+import java.util.Map;
+import javax.annotation.Nonnull;
+
+
 /**
  * Interface for configs of {@link RecordExtractor}
  */
 public interface RecordExtractorConfig {
+  default void init(@Nonnull Map<String, String> props) { }
```

Comment:
(convention) We don't usually annotate `non-null`, but only annotate `nullable`. Also make it multiple lines
```suggestion
  default void init(Map<String, String> props) {
  }
```